src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsClassifier.cs

```diff
@@ -223,14 +223,37 @@ public bool Matches (MethodDefinition method)
 						parameterTypeName = pd.ParameterType.FullName;
 					}
 
-					if (String.Compare (parameterTypeName, paramTypes[i], StringComparison.Ordinal) != 0) {
+					if (!TypeMatches (parameterTypeName, paramTypes[i])) {
 						log.LogWarning ($"Method '{method.FullName}' doesn't match native callback signature, expected parameter type '{paramTypes[i]}' at position {i}, found '{parameterTypeName}'");
 						return false;
 					}
 				}
 
 				return true;
 			}
+
+			// Because these types are marshaled as different blittable types,
+			// we need to accept them as equivalent
+			static readonly Tuple<string, string>[] equivalent_types = new [] {
```

Comment:
I would prefer the use of "named `ValueTuple<…>`":

```csharp
static readonly (string Source, string Replacement)[] equivalent_types = new[]{
    (Source: "System.Boolean", Replacement: "System.SByte"),
    (Source: "System.Char", Replacement: "System.UInt16"),
};
```

Then `TypeMatches()` can use `.Source` instead of `.Item1`, etc.